subprojects/docs/src/docs/userguide/moreAboutTasks.adoc

```diff
@@ -580,6 +580,49 @@ Input properties annotated with `@api:org.gradle.api.tasks.CompileClasspath[]` a
 Additionally to the aforementioned general classpath rules, compile classpaths ignore changes to everything but class files. Gradle uses the same class analysis described in <<sec:java_compile_avoidance>> to further filter changes that don't affect the class' ABIs.
 This means that changes which only touch the implementation of classes do not make the task out of date.
 
+[[sec:task_input_nested_inputs]]
+====== Nested inputs
```

Comment:
@lptr @wolfs Are there any example uses of `@Nested` we can provide or link to in the user manual or guides? Right now I see only a couple mentions of `@Nested` but no information about how to actually use it.

Reply:
The newest one is the one in `CompileOptions`: https://github.com/gradle/gradle/blob/252f30e5efba8099450e28d5e2c0390413b9055f/subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java#L363

This class contains quite some more `@Nested` inputs.

Moreover, [`JavaCompile.options`](https://github.com/gradle/gradle/blob/b66e4d0664c49215661a061ba5f5d87d77f54dbb/subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java#L231) itself is a nested property. 

It is also used for the code quality plugins, e.g. [here](https://github.com/gradle/gradle/blob/9ab5b27736026316cf1a84932dc227cab25ac611/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java#L274).

Or did you mean something else?

Reply:
Specifically, I am hoping for an example we can embed within docs. I suppose the minimum viable solution here would be to link to our sources that use this, but better for us to have a small complete example with explanations.